### PR TITLE
[WIP] Add a test showing providing external connection for a send-only endpoint

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

File: src/NServiceBus.SqlServer.TransportTests/NServiceBus.SqlServer.TransportTests.csproj [hunk 1]
```diff
@@ -16,4 +16,8 @@
     <PackageReference Include="NUnit3TestAdapter" Version="3.8.0-alpha1" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
```
Comment (MarcinHoppe) [inline src/NServiceBus.SqlServer.TransportTests/NServiceBus.SqlServer.TransportTests.csproj]: What's this?

File: src/NServiceBus.SqlServer.AcceptanceTests/When_providing_external_connection.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                        transportTransaction.Set(transaction);
+
+                        var options = new SendOptions();
+                        options.GetExtensions().Set(transportTransaction);
+                        await bus.Send(new Message(), options);
+
+                        transaction.Commit();
+                    }
+                }))
+                .Done(c => c.MessageReceived)
```
Comment (MarcinHoppe) [inline src/NServiceBus.SqlServer.AcceptanceTests/When_providing_external_connection.cs]: How do we know if the transport used the provided connection and transaction? Do we need more assertions here? Maybe rolling back the transaction after `bus.Send` would be a better indication that the same transaction is used?

Comment (SzymonPobiega) [reply src/NServiceBus.SqlServer.AcceptanceTests/When_providing_external_connection.cs]: True. I forgot that the connection string is set by the test infrastructure.

File: src/NServiceBus.SqlServer.TransportTests/NServiceBus.SqlServer.TransportTests.csproj (hunk 1 above)
Comment (SzymonPobiega) [reply src/NServiceBus.SqlServer.TransportTests/NServiceBus.SqlServer.TransportTests.csproj]: VisualStudio adds this crap automatically. I'll remove.

**Pull request closed without merging**